scripts/eda_kube.sh

```diff
@@ -0,0 +1,118 @@
+#!/usr/bin/env bash
+set -o errexit
+set -o pipefail
+set -o nounset
+
+SCRIPTS_DIR=$( cd -- "$( dirname -- "${BASH_SOURCE[0]}" )" &> /dev/null && pwd )
+
+CMD=${1:-help}
+UI_LOCAL_PORT=${2:-''}
+
+export DEBUG=${DEBUG:-false}
+
+# import common & logging
+source "$SCRIPTS_DIR"/common/logging.sh
+source "$SCRIPTS_DIR"/common/utils.sh
+
+trap handle_errors ERR
+
+handle_errors() {
+  log-err "An error occurred on or around line ${BASH_LINENO[0]}. Unable to continue."
+  exit 1
+}
+
+# deployment templates dir
+TEMPLATES_DIR="$SCRIPTS_DIR"/templates
+
+# deployment dir
+DEPLOY_DIR="$SCRIPTS_DIR"/../tools/deploy
+
+# docker dir
+DOCKER_FILE="$SCRIPTS_DIR"/../tools/docker/Dockerfile
+
+# default local ports for forwarding
+AE_UI_FWD_PORT=${2:-8080}
+
+usage() {
+    log-info "Usage: $(basename "$0") <command> [command_arg]"
+    log-info ""
+    log-info "commands:"
+    log-info "\t build                         build image and create deployment"
+    log-info "\t deploy                        deploy to k8s instance (minikube, etc...)"
+    log-info "\t help                          show usage"
+    log-info "\t port-forward-ui <port>        forward local port to Events UI (default: 8080)"
+}
+
+help() {
+    usage
+}
+
+build-deployment() {
+  log-info "Deployment: $DEPLOY_DIR/temp"
+  log-info "\tIMAGE: $AE_UI_IMAGE:latest"
+
+  log-debug "kustomize edit set image ansible-events-ui=${AE_UI_IMAGE}:latest"
+  cd "$DEPLOY_DIR"/events-ui
+  kustomize edit set image "ansible-events-ui=${AE_UI_IMAGE}:latest"
+  cd ../../../
+  log-debug "kustomize build ./tools/deploy -o ./tools/deploy/temp"
+  kustomize build ./tools/deploy -o ./tools/deploy/temp
+}
+
+build() {
+    local _version=$(git rev-parse --short=7 HEAD)
+
+    log-info "$CONTAINER_EXEC build . -t  ${AE_UI_IMAGE}:${_version} -f $DOCKER_FILE"
+    $CONTAINER_EXEC build . -t  "${AE_UI_IMAGE}":"${_version}" -f "$DOCKER_FILE"
+
+    log-info "$CONTAINER_EXEC tag ${AE_UI_IMAGE}:${_version} ${AE_UI_IMAGE}:latest"
+    $CONTAINER_EXEC tag "${AE_UI_IMAGE}":"${_version}" "${AE_UI_IMAGE}":latest
+
+    log-info "$CONTAINER_EXEC push  ${AE_UI_IMAGE}:${_version}"
+    $CONTAINER_EXEC push "${AE_UI_IMAGE}":"${_version}"
+    $CONTAINER_EXEC push "${AE_UI_IMAGE}":latest
+
+    build-deployment
```

Comment:
This cmd belongs to the deploy task.

Reply:
The reason for the separation of these two functions is for future needs.

Reply:
I meant that `build-deployment` call is for the deployment not for the image building/pushing, and it should be in that function.